lib/utils/queryable.js

```diff
@@ -113,12 +113,14 @@ const getter = ({ data, key }) => {
     } else {
       // if can't find any more values, it means it's just over
       // and there's nothing to return
-      if (!_data[k]) {
+      if (Object.hasOwn(_data, k) && !_data[k]) {
```

Comment:
The falsey check here was a bad idea in the first place.  What if the value of `node-gyp` is `false`?  Shouldn't I get that back?

Reply:
I initially didn't have that check there, but without it a bunch of other tests fail.

![Screenshot 2023-10-16 at 10 46 20 AM](https://github.com/npm/cli/assets/2738518/7c49a3e1-9a54-4d80-89a4-8c45bf607f4c)

Reply:
Ah nevermind, after I run the `npm run snap` it seems like those fails clear up. But I need to add another test it seems since it now doesn't meet the 100% coverage (99.61 currently).

Reply:
I removed this check and replaced it with explicit checks for boolean and undefined.